pkg/cli/admin/upgrade/status/mco/layered_node_state.go

```diff
@@ -228,39 +240,35 @@ func isNodeMCDState(node *corev1.Node, state string) bool {
 	return dstate == state
 }
 
-func checkNodeReady(node *corev1.Node) error {
+func checkNodeReady(l *LayeredNodeState) error {
+	node := l.node
 	for i := range node.Status.Conditions {
 		cond := &node.Status.Conditions[i]
 		// We consider the node for scheduling only when its:
 		// - NodeReady condition status is ConditionTrue,
 		// - NodeDiskPressure condition status is ConditionFalse,
 		// - NodeNetworkUnavailable condition status is ConditionFalse.
 		if cond.Type == corev1.NodeReady && cond.Status != corev1.ConditionTrue {
+			l.ReasonOfUnavailability = fmt.Sprintf("node is reporting NotReady=%v with message=%q", cond.Status, cond.Message)
```

Comment:
I know we duplicate the error message below but our new strings will be seen by the users - I wonder if we should use identifiers and language that they are more likely to be familiar with? These are node conditions and their names are different - they are presented as such in both web console and `oc describe` output.  `NotReady` and `OutOfDisk` names may be reasons? Not sure. But the message will also be incorrect. We will typically get into this branch when Ready=False, but the message will actually say "...reporting NotReady=False..."
 
Also, because these come from conditions, they have `LastTransitionTime` which would allow us to set a `since` for our insight, which we otherwise do not have for unavailable insight - that would be great to have!